bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -480,9 +517,25 @@ public String toString() {
      */
     public void asyncReadEntries(long firstEntry, long lastEntry,
                                  ReadCallback cb, Object ctx) {
+
         // Little sanity check
-        if (firstEntry < 0 || lastEntry > lastAddConfirmed
-                || firstEntry > lastEntry) {
+        if (firstEntry < 0 || firstEntry > lastEntry) {
+            cb.readComplete(BKException.Code.ReadException, this, null, ctx);
```

Comment:
@sijie I believe we should make the error code more granular here. Will you be ok with the following change?

if (firstEntry < 0 || firstEntry > lastEntry) {
    cb.readComplete(BKException.Code.IncorrectParameterException)
}
if ( lastEntry > lastAddConfirmed ) {
    cb.readComplete(BKException.Code.ReadException)
}

I think this change makes lot of sense to return correct error back. Do you see any issues with this??